consensus/src/simplex/types.rs

```diff
@@ -50,6 +50,52 @@ pub trait Attributable {
     fn signer(&self) -> u32;
 }
 
+/// AttributableVec is a vector of [Attributable] items where a given [Attributable::signer()]
+/// can add at most one item.
+pub struct AttributableVec<T: Attributable> {
+    data: Vec<Option<T>>,
+    added: usize,
```

Comment:
The alternative here would be to use a BitMap (I actually think that may be dramatically more efficient because we will only get the minimum required votes in Voter).

Reply:
I concluded a single upfront allocation (even if not always full) is better than trying to regrow the vector as items are added (we'll usually fill up).

Reply:
I think what I'm doing now is more efficient (albeit very minorly) than maintaining a BitVec with this (given we would already be preallocating the vec).